ReactiveCocoa/DelegateProxy.swift

```diff
@@ -53,36 +54,46 @@ extension DelegateProxy {
 	internal static func proxy<P: DelegateProxy<Delegate>>(
 		for instance: NSObject,
 		setter: Selector,
-		getter: Selector,
-		_ key: StaticString = #function
+		getter: Selector
 	) -> P {
-		return _proxy(for: instance, setter: setter, getter: getter, key) as! P
+		return _proxy(for: instance, setter: setter, getter: getter) as! P
 	}
 
 	private static func _proxy(
 		for instance: NSObject,
 		setter: Selector,
-		getter: Selector,
-		_ key: StaticString = #function
+		getter: Selector
 	) -> AnyObject {
-		let key = AssociationKey<DelegateProxy<Delegate>?>(key)
-
 		return instance.synchronized {
-			if let proxy = instance.associations.value(forKey: key) {
+			let proxyMap = instance.associations.value(forKey: proxyMapKey)
+
+			if let proxy = proxyMap.first(where: { $0.0 == setter })?.1 {
 				return proxy
 			}
 
 			let newSetterImpl: @convention(block) (NSObject, AnyObject?) -> Void = { object, delegate in
-				let proxy = object.associations.value(forKey: key)!
-				proxy.forwardee = (delegate as! Delegate?)
+				let proxyMap = object.associations.value(forKey: proxyMapKey)
+
+				if let proxy = proxyMap.first(where: { $0.0 == setter })?.1 as! DelegateProxy<Delegate>? {
+					proxy.forwardee = (delegate as! Delegate?)
+				} else {
+					typealias Setter = @convention(c) (NSObject, Selector, AnyObject?) -> Void
+					let impl = class_getMethodImplementation(object.objcClass, setter)
+					let delegateSetter = unsafeBitCast(impl, to: Setter.self)
+					delegateSetter(object, setter, delegate)
+				}
 			}
 
 			// Hide the original setter, and redirect subsequent delegate assignment
 			// to the proxy.
 			instance.swizzle((setter, newSetterImpl), key: hasSwizzledKey)
 
-			// Set the proxy as the delegate.
-			let realClass: AnyClass = class_getSuperclass(object_getClass(instance))
+			// Retrieve the original delegate setter.
+			//
+			// `instance.objcClass` (aka `+class`) would report the perceived
+			// class instead of the real runtime subclass generated by RAC or
+			// KVO.
+			let realClass: AnyClass = instance.objcClass
```

Comment:
While in most cases the superclass of the real runtime class is the original/perceived class, it is not true if the instance is first swizzled by RAC then by KVO. It would end up with being `NSKVONotifying_*_RACSwift`, i.e. two levels of inheritance.